compiler/test/fail_compilation/fail6795.d

```diff
@@ -2,27 +2,27 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/fail6795.d(19): Error: cannot modify expression `[0][0]` because it is not an lvalue
-fail_compilation/fail6795.d(20): Error: cannot modify expression `[0:0][0]` because it is not an lvalue
-fail_compilation/fail6795.d(22): Error: cannot modify expression `[0][0]` because it is not an lvalue
-fail_compilation/fail6795.d(23): Error: cannot modify expression `[0:0][0]` because it is not an lvalue
-fail_compilation/fail6795.d(25): Error: cannot take address of expression `[0][0]` because it is not an lvalue
-fail_compilation/fail6795.d(26): Error: cannot take address of expression `[0:0][0]` because it is not an lvalue
-fail_compilation/fail6795.d(30): Error: cannot modify expression `Some["zz"]` because it is not an lvalue
+fail_compilation/fail6795.d(16): Error: cannot modify expression `[0][0]` because it is not an lvalue
+fail_compilation/fail6795.d(20): Error: cannot modify expression `[0][0]` because it is not an lvalue
+fail_compilation/fail6795.d(24): Error: cannot take address of expression `[0][0]` because it is not an lvalue
 ---
 */
 void test_wrong_line_num()
 {
     enum int[1] sa = [0];
-    enum int[int] aa = [0:0];
+    enum int[] da = [0];      // dynamically allocated whenever accessed, so modifiable
+    enum int[int] aa = [0:0]; // ditto
```

Comment:
@dkorpel The change from https://github.com/dlang/dmd/pull/17094 is partially undone here (it still does not crash, which was the actual bug report). Other operations on the associative array still allow modification, e.g.  via `p = 0 in aa`, so why not this one? I would assume an AA to be closer in behavior to a dynamic array than to a static array. As the latter also creates a copy on the heap, it is also debatable why it is not allowed as an lvalue. Assigning to it with `sa.ptr[0] = 1` even works.